### PR TITLE
FFTX Update

### PR DESCRIPTION
**Pull request by broderickpt**
Update recipes for packages: fftx, spiral-software, and spiral-package-xxxx (xxxx = { fftx, simt, hcol, mpi, jit }
Due to 'spack activate' being no longer supported change spiral-software to add variants for each spiral-package.
spiral-software must now be installed with the variants required (i.e., spiral-software will depend on the packages requested, which is the inverse of the prior dependencies).

@scheibelp does this look ok?

### CODE REVIEW

**Comment by spiralgen** [comment]
This PR looks all good

File: var/spack/repos/builtin/packages/fftx/package.py [hunk 1]
```diff
@@ -11,20 +13,22 @@ class Fftx(CMakePackage, CudaPackage, ROCmPackage):
...
     version("develop", branch="develop")
     version("main", branch="main")
+    version("1.1.2", sha256="b2c4a7791305481af9e1bd358c1215efa4506c91c943cddca3780a1ccbc27810")
+    version("1.1.1", sha256="5cbca66ef09eca02ee8f336f58eb45cfac69cfb29cd6eb945852ad74085d8a60")
+    version("1.1.0", sha256="a6f95605abc11460bbf51839727a456a31488e27e12a970fc29a1b8c42f4e3b5")
     version("1.0.3", sha256="b5ff275facce4a2fbabd0aecc65dd55b744794f2e07cd8cfa91363001c664896")
 
     depends_on("spiral-software")
-    depends_on("spiral-package-fftx")
-    depends_on("spiral-package-simt")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/fftx/package.py:26]: Based on these two removed `depends_on` declarations, the above `depends_on(spiral-software)` should now be `depends_on(spiral-software+fftx+simt)` (i.e., the `spiral-software` package this depends on should always be built with `+fftx` and `+simt`).

If that `spiral-software` instance should also have `+hcol`, `+mpi`, `+jit`, then those should be specified here too. If they are optional , then it is fine to leave them off.

**Review by scheibelp — COMMENTED** [review]
I see only one potential issue: let me know if this seems off-track.

File: var/spack/repos/builtin/packages/fftx/package.py (hunk 1 above)
Comment (broderickpt) [reply var/spack/repos/builtin/packages/fftx/package.py:26]: Yes, good point.  I changed the depends_on() to read: ```depends_on("spiral-software+fftx+simt+jit+mpi")```.  There is no need for hcol, so I left it off.

**Review by scheibelp — APPROVED** [review]

**Pull request merged**